packages/core/src/service/services/order.service.ts

```diff
@@ -359,10 +359,14 @@ export class OrderService {
         customFields?: { [key: string]: any },
     ): Promise<ErrorResultUnion<UpdateOrderItemsResult, Order>> {
         const order = await this.getOrderOrThrow(ctx, orderId);
+        const orderLineByVariant = order.lines.find(line =>
```

Comment:
I think this line will fail when OrderLines exist with the same ProductVariant but different custom fields. This is what the `getExistingOrderLine` method handles. You could move the `existingOrderLine` call up above the validation.